src/components/CheckboxInput/CheckboxInput.tsx

```diff
@@ -9,93 +9,72 @@ import './CheckboxInput.scss';
  * Used to allow users to make a range of selections (zero, one or many).
  */
 
-const propTypes = {
+interface Props {
   /**
    * The id attribute of the input
    */
-  id: PropTypes.string.isRequired,
+  id: string;
   /**
    * Additional classes to add
    */
-  className: PropTypes.string,
+  className?: string;
   /**
    * Mark the input field as invalid and display a validation message.
    * Pass a string or node to render a validation message below the input
    */
-  error: PropTypes.oneOfType([
-    PropTypes.bool,
-    PropTypes.string,
-    PropTypes.node,
-  ]),
+  error?: React.ReactNode;
   /**
    * The checkbox input "checked" attribute
    */
-  isChecked: PropTypes.bool,
+  isChecked?: boolean;
   /**
    * If the input should be disabled and not focusable
    */
-  isDisabled: PropTypes.bool,
+  isDisabled?: boolean;
   /**
    * Determines if input is required or not. (Label will have an asterisk if required)
    */
-  isRequired: PropTypes.bool,
+  isRequired?: boolean;
   /**
    * Callback function when input is blurred.
    */
-  onBlur: PropTypes.func,
+  onBlur?: (event: React.FocusEvent<HTMLInputElement>) => void;
   /**
    * Callback function when input is changed
    */
-  onChange: PropTypes.func.isRequired,
+  onChange?: (event: ChangeEvent<HTMLInputElement>) => void;
   /**
    * Callback function when input is focused
    */
-  onFocus: PropTypes.func,
+  onFocus?: (event: FocusEvent<HTMLInputElement>) => void;
   /**
    * Custom content to be displayed to right of checkbox. Can be any valid node/tree, anchors, etc.
    */
-  label: PropTypes.node.isRequired,
+  label?: string;
```

Comment:
Was a required node before, now an optional string?

Reply:
i can change it back to required, but it was weird that it was required given that we do a check for it when rendering.

it needs to be a string because it's passed into aria-label, which doesnt accept react nodes